sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -95,25 +95,32 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
    * {{{
    *   ANALYZE TABLE table COMPUTE STATISTICS [NOSCAN];
    * }}}
+   * Example SQL for analyzing a single partition :
+   * {{{
+   *   ANALYZE TABLE table PARTITION (key=value,..) COMPUTE STATISTICS [NOSCAN];
```

Comment:
The existing syntax is not good. Could you improve them?
```SQL
ANALYZE TABLE [db_name.]tablename [PARTITION(partcol1[=val1], partcol2[=val2], ...)] 
COMPUTE STATISTICS [NOSCAN]
```

In addition, since we have a restriction, please do not call `visitNonOptionalPartitionSpec`. Instead, we can capture the non-set partition column and issue a more user-friendly exception message.